Tests/SolanaTests/Models/TransactionTest.swift

```diff
@@ -0,0 +1,194 @@
+        XCTAssertEqual(message.accountKeys[9].publicKey, programId)
+    }
+    
+    @available(iOS 13.0.0, *)
+    func testAccountKeysCollapsesSignednessAndWritebility() async throws {
+        // These pubkeys are chosen specially to be in sort order.
+        let payer = PublicKey(string:
+            "2eBgaMN8dCnCjx8B8Wrwk974v5WHwA6Vvj4N2mW9KDyt"
+        )!
+        let account2 = PublicKey(
+            string: "DL8FErokCN7rerLdmJ7tQvsL1FsqDu1sTKLLooWmChiW"
+        )!
+        let account3 = PublicKey(
+            string: "EdPiTYbXFxNrn1vqD7ZdDyauRKG4hMR6wY54RU1YFP2e"
+        )!
+        let account4 = PublicKey(
+            string: "FThXbyKK4kYJBngSSuvo9e6kc7mwPHEgw4V8qdmz1h3k"
+        )!
+        let programId = PublicKey(
+            string: "Gcatgv533efD1z2knsH9UKtkrjRWCZGi12f8MjNaDzmN"
+        )!
+        let account5 = PublicKey(
+            string: "rBtwG4bx85Exjr9cgoupvP1c7VTe7u5B36rzCg1HYgi"
+        )!
+
+        let recentBlockhash = HotAccount()!.publicKey.base58EncodedString
+        
+        let accountMetas: [AccountMeta] = [
+            // Should sort last.
+            AccountMeta(publicKey: account5, isSigner: false, isWritable: false),
+            AccountMeta(publicKey: account5, isSigner: false, isWritable: false),
+            // Should be considered writeable.
+            AccountMeta(publicKey: account4, isSigner: false, isWritable: true),
+            AccountMeta(publicKey: account4, isSigner: false, isWritable: true),
+            // Should be considered a signer.
+            AccountMeta(publicKey: account3, isSigner: true, isWritable: false),
+            AccountMeta(publicKey: account3, isSigner: true, isWritable: false),
+            // Should be considered a writable signer.
+            AccountMeta(publicKey: account2, isSigner: true, isWritable: true),
+            AccountMeta(publicKey: account2, isSigner: true, isWritable: true),
+            // Payer
+            AccountMeta(publicKey: payer, isSigner: true, isWritable: true),
+        ]
+        
+        let transactionIntruction = TransactionInstruction(keys: accountMetas, programId: programId, data: [])
+        let transaction = Transaction(feePayer: payer, instructions: [transactionIntruction], recentBlockhash: recentBlockhash)
+        
+        let message = try transaction.compileMessage().get()
+
+        // Payer comes first.
+        XCTAssertEqual(message.accountKeys[0].publicKey, payer)
+
+        // Writable signer comes first.
+        XCTAssertEqual(message.accountKeys[1].publicKey, account2)
+
+        // Signer comes next.
+        XCTAssertEqual(message.accountKeys[2].publicKey, account3)
+
+        // Writable account comes next.
+        XCTAssertEqual(message.accountKeys[3].publicKey, account4)
+
+        // Regular accounts come last.
+        XCTAssertEqual(message.accountKeys[4].publicKey, programId)
+        XCTAssertEqual(message.accountKeys[5].publicKey, account5)
+    }
+    
+    @available(iOS 13.0.0, *)
```

Comment:
And again. Actually, you could put the `@available` on the whole class, rather than on each individual function

Reply:
Moved to the class level.